ui/src/views/Documentation/Reference/SchemaPanel.jsx

```diff
@@ -0,0 +1,144 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Grid from '@material-ui/core/Grid';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import SchemaTable from '../../../components/SchemaTable';
+
+const primaryTypographyProps = { variant: 'body1' };
+
+@withRouter
+@withStyles(
+  () => ({
+    expansionPanelSummary: {
+      margin: 0,
+    },
+    list: {
+      width: '100%',
+    },
+    gridContainer: {
+      '& > div': {
+        margin: 'auto',
+      },
+    },
+  }),
+  { withTheme: true }
+)
+export default class SchemaPanel extends Component {
+  static propTypes = {
+    /** The reference entry. */
+    entry: object.isRequired,
+    /** The service name in which the entry belongs to. */
+    serviceName: string.isRequired,
+  };
+
+  state = {
+    expanded: false,
+  };
+
+  renderExchangeExpansionPanelSummary = () => {
+    const { entry, classes } = this.props;
+
+    return (
+      <Grid className={classes.gridContainer} container spacing={8}>
+        <Grid item xs={5}>
+          <div>
+            <Typography id={entry.content.$schema} component="h3">
+              {entry.content.title
+                ? entry.content.title
+                : entry.content.$schema}
+            </Typography>
+          </div>
+        </Grid>
+        <Grid item xs={7}>
+          <div>
+            <Typography>
+              {entry.content.$id ? entry.content.$id : entry.content.$schema}
+            </Typography>
+          </div>
+        </Grid>
+      </Grid>
+    );
+  };
+
+  renderSchemaTable = (schema, headerTitle) => {
+    const { serviceName } = this.props;
+
+    return (
+      <ListItem>
+        <ListItemText
+          primaryTypographyProps={primaryTypographyProps}
+          disableTypography
+          primary={<Typography variant="body1">{headerTitle}</Typography>}
+          secondary={
+            <Fragment>
+              <br />
+              <SchemaTable schema={schema} serviceName={serviceName} />
+            </Fragment>
+          }
+        />
+      </ListItem>
+    );
+  };
+
+  renderExchangeExpansionDetails = () => {
+    const { classes, entry } = this.props;
+    const { expanded } = this.state;
+
+    return (
+      expanded && (
+        <List className={classes.list}>
+          {this.renderSchemaTable(entry.content.$id, entry.content.description)}
+        </List>
+      )
+    );
+  };
+
+  handlePanelChange = key => () => {
+    const { entry, history } = this.props;
+    const { expanded } = this.state;
+
+    if (window.location.hash === `#${entry.content[key]}` || expanded) {
+      history.push(history.location.pathname);
+    } else {
+      history.push(`#${entry.content[key]}`);
+    }
+
+    this.setState({
+      expanded: !expanded,
+    });
+  };
+
+  render() {
+    const { classes, entry } = this.props;
+    const { expanded } = this.state;
+    const entryHashKey = entry.content.$id ? '$id' : '$schema';
```

Comment:
Are there schemas that don't have `$id`?  There *really* shouldn't be :)

Reply:
Yes, there are a couple of schemas which does not have `$id`

`Exchanges to manage hooks` this is one of the example

Reply:
Ah, that's not a schema -- it's a reference document.  Sorry, I forgot the format of `references.json` was confusing.  It's a list of objects of the form `{filename, content}`, and contains both schemas and references.  The schemas can be identified by having filenames beginning with `schemas/`.  For this view, the references can be ignored.

Reply:
So, if something does not have `$id`, we will ignore those objects from the view?

Reply:
Anything that has a filename beginning with `schemas/` will definitely have an `$id`.  There's another data format that I was thinking of that looks like `{references: [..], schemas: [..]}`, and everything under the `schemas` property there is a schema and thus has an `$id`.  That's why I didn't mention this earlier.  Sorry about that!